drivers/media/i2c/imx477.c

```diff
@@ -1927,7 +1967,8 @@ static int imx477_init_controls(struct imx477 *imx477)
 	imx477->exposure = v4l2_ctrl_new_std(ctrl_hdlr, &imx477_ctrl_ops,
 					     V4L2_CID_EXPOSURE,
 					     IMX477_EXPOSURE_MIN,
-					     IMX477_EXPOSURE_MAX,
+					     IMX477_EXPOSURE_MAX *
+					       (1 << IMX477_LONG_EXP_SHIFT_MAX),
```

Comment:
I haven't compiled or run this, but doesn't this leave a max exposure that can't be achieved under most conditions. Shouldn't the exposure range (and step?) be updated when a VBLANK change results in a change of the multiplier?

Ah we do adjust it via ```imx477_adjust_exposure_range```, and that is called at the end of this function. Seeing as the default VBLANK is going to have a multiplier of x1 I'd be tempted to leave this alone, even if it is then updated at the end of the function.

Reply:
Yes, it will be updated correctly so no harm in leaving this as the orginal.